docs/100-upgrade/migration-guides/3.2-3.3.mdx

```diff
@@ -0,0 +1,109 @@
+  to 3.3.
+---
+
+<p>{frontMatter.description}</p>
+
+## Update dependencies
+
+Update all your `@front-commerce/*` dependencies to this version:
+
+```shell
+pnpm update "@front-commerce/\*@3.3.0"
+```
+
+## Code changes
+
+### Remix entrypoints
+
+In this release, we added new features that required to hook into your Remix
+application entrypoints created with the initial skeleton.<br /> You must update
+these files in your application, as detailed below **or copy the ones from the
+latest skeleton**.
+
+<details>
+  <summary><code>root.tsx</code> file</summary>
+
+```diff
+diff --git a/skeleton/app/root.tsx b/skeleton/app/root.tsx
+index a28dcf87e..9030e925f 100644
+--- a/skeleton/app/root.tsx
++++ b/skeleton/app/root.tsx
+@@ -37,14 +37,7 @@ export const loader = async ({ context }: LoaderFunctionArgs) => {
+   return json({
+     shop: response.shop,
+     device: app.config.device,
+-    process: {
+-      env: {
+-        NODE_ENV: process.env.NODE_ENV,
+-        SERVER: false,
+-        FRONT_COMMERCE_WEB_DEV_ANALYTICS_WARNING_DISABLE:
+-          process.env.FRONT_COMMERCE_WEB_DEV_ANALYTICS_WARNING_DISABLE,
+-      },
+-    },
++    publicConfig: app.config.public,
+   });
+ };
+```
+
+</details>
+
+### Deprecate `process.env.FRONT_COMMERCE_WEB_*` usage
+
+**Breaking:** In this release, we updated the way we handle public
+configuration.
+
+To cope with these changes, you will need to ensure that you replace any of
+these usages with the new `publicConfig` values. We introduced a `3.3.0` codemod
+to automatically update your codebase, follow the
+[Automated Migration](#automated-migration) to run the codemod.
+
+Before:
+
+```js
+const {
+  FRONT_COMMERCE_WEB_DEV_ANALYTICS_WARNING_DISABLE,
+  FRONT_COMMERCE_WEB_DEV_DISABLE_PASSWORD_HINT,
+} = process.env;
+
+const example = {
+  analyticsWarningDisable: FRONT_COMMERCE_WEB_DEV_ANALYTICS_WARNING_DISABLE,
+  disablePasswordHint: FRONT_COMMERCE_WEB_DEV_DISABLE_PASSWORD_HINT,
+  myCustomValue: process.env.FRONT_COMMERCE_WEB_MY_CUSTOM_VALUE,
+};
+```
+
+After:
+
+```js
+import { getPublicConfig } from "@front-commerce/core/react";
+
+// After running your application for the first time,
+// you should have all typed values in the public config
+const publicConfig = getPublicConfig();
+
+const example = {
+  analyticsWarningDisable: publicConfig?.analytics?.disableDevWarning ?? false,
+  disablePasswordHint: publicConfig?.password?.disablePasswordHint ?? false,
+  myCustomValue: getPublicConfig("FRONT_COMMERCE_WEB_MY_CUSTOM_VALUE"), // This will warn with a deprecation message
+};
+```
+
+:::caution
+
+We highly suggest that you move any of your custom values to a new config
+provider which extends the `public` schema.
+
+See [Public configurations documentation](/docs/3.x/guides/public-configuration)
+for more details.
+
+:::
+
+## Automated Migration
+
+```shell
+pnpm run front-commerce migrate --transform 3.3.0
+```
```

Comment:
**thought:** it looks like a step we'll have during every release. Shouldn't we move this before each specific section? (i.e: right after the dependency update section)

Reply:
It makes sense yes.